### PR TITLE
[@wordpress/components] Add CustomSelectControl

### PR DESCRIPTION
**Pull request by sarayourfriend**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## wordpress__components ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__components@latest/))
was missing the following properties:
1. Guide
1. GuidePage
1. TextHighlight
1. Tip
1. ToolbarGroup

<details>
<summary>as well as these 1 other properties...</summary>
<p>VisuallyHidden</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 60f88cbd1ebb184b45501b3ea285036b733584a2
</p>

File: types/wordpress__components/package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "private": true,
     "dependencies": {
         "@wordpress/element": "^2.14.0",
+        "downshift": "^5.4.0",
```
Comment (sirreal) [inline types/wordpress__components/package.json:5]: It does seem to bundle typings:

https://unpkg.com/browse/downshift@5.0.4/typings/index.d.ts

This version aligns with the latest `@wordpress/components` release:

https://unpkg.com/browse/@wordpress/components@11.0.0/package.json

**Review by sirreal — APPROVED** [review]
Thanks, makes sense and looks right 👍

**Comment by saramarcondes** [comment]
Ready to merge

File: types/wordpress__components/package.json (hunk 1 above)
Comment (rbuckton) [reply types/wordpress__components/package.json:5]: The version header for the index.d.ts for this package is `// Type definitions for @wordpress/components 9.8`. The version of downshift used for v9.8 was `^4.0.5`. I would recommend either:

- Using the typings from downshift@4.0.5, or
- Updating the version header in index.d.ts

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Review by sirreal — APPROVED** [review]

**Comment by orta** [comment]
Cool, thanks @sirreal - merging

**Pull request merged**